crates/re_ui/src/drag_and_drop.rs

```diff
@@ -0,0 +1,309 @@
+//! Helpers for drag and drop support. Works well in combination with [`crate::list_item::ListItem`].
+
+/// Context information related to a candidate drop target, used by [`find_drop_target`] to compute the [`DropTarget`],
+/// if any.
```

Comment:
What constitute a drop target?

For a list of items:
* Is each item a drop target?
* Is the upper half and lower half of each item a drop target (before/after)?
* Is the drop target between the items (insertion points)?

Reply:
None of those things actually. That structure is basically just context information about the hovered item, which is used by `find_drop_target()` to formulate a "logical" drop target (container ID + position) based on actual mouse position.

I renamed that structure to `ItemContext` and added some high-level description of what's going in `find_drop_target()`'s docstring.

I also changed the signature to pass rectangle instead of responses, to reflect the purely "geometric" scope of the function.